lib/permutations.rb

```diff
@@ -1,4 +1,20 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented"
+  hash = {}
+  string1.each do |letter|
+    if hash[letter]
+      hash[letter] += 1
+    else
+      hash[letter] = 1
+    end
+  end
+  
+  string2.each do |letter|
+    if hash[letter] && hash[letter] != 0
+      hash[letter] -= 1
+    else
+      return false
+    end
+    return true
```

Comment:
This return should be **after** the loop, and should only be true if all the values in the hash are now 0.